common/src/main/scala/org/apache/celeborn/common/util/ThreadUtils.scala

```diff
@@ -44,16 +45,23 @@ object ThreadUtils {
   /**
    * Create a thread factory that names threads with a prefix and also sets the threads to daemon.
    */
-  def namedThreadFactory(prefix: String): ThreadFactory = {
-    new ThreadFactoryBuilder().setDaemon(true).setNameFormat(prefix + "-%d").build()
+  def namedThreadFactory(prefix: String, logger: Logger): ThreadFactory = {
+    new ThreadFactoryBuilder()
+      .setDaemon(true)
+      .setNameFormat(prefix + "-%d")
+      .setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler {
+        override def uncaughtException(t: Thread, e: Throwable): Unit =
+          logger.error("Uncaught exception in executor service {}, thread {}", prefix, t, e)
```

Comment:
as there is a "prefix" in the error message, which could tell us where the thread comes from, I don't think we could benefit more from the logger name.

Reply:
could you apply the same change to `def newDaemonSingleThreadExecutor` and `newDaemonSingleThreadScheduledExecutor`?